Index: gcc-4.5.2/gcc/config/sol2.h
===================================================================
--- gcc-4.5.2/gcc/config/sol2.h
+++ gcc-4.5.2/gcc/config/sol2.h
@@ -156,12 +156,12 @@ along with GCC; see the file COPYING3.  
    %{YP,*} \
    %{R*} \
    %{compat-bsd: \
-     %{!YP,*:%{p|pg:-Y P,%R/usr/ucblib:%R/usr/ccs/lib/libp:%R/usr/lib/libp:%R/usr/ccs/lib:%R/usr/lib} \
-             %{!p:%{!pg:-Y P,%R/usr/ucblib:%R/usr/ccs/lib:%R/usr/lib}}} \
-             -R %R/usr/ucblib} \
+     %{!YP,*:%{p|pg:-Y P,/usr/ucblib:/usr/ccs/lib/libp:/usr/lib/libp:/usr/ccs/lib:/lib:/usr/lib:/usr/gnu/lib:/usr/gcc/4.5/lib -R /usr/ucblib:/usr/ccs/lib/libp:/usr/lib/libp:/usr/ccs/lib:/lib:/usr/lib:/usr/gnu/lib:/usr/gcc/4.5/lib} \
+             %{!p:%{!pg:-Y P,/usr/ucblib:/usr/ccs/lib:/lib:/usr/lib:/usr/gnu/lib:/usr/gcc/4.5/lib -R /usr/ucblib:/usr/ccs/lib:/lib:/usr/lib:/usr/gnu/lib:/usr/gcc/4.5/lib}}} \
+             } \
    %{!compat-bsd: \
-     %{!YP,*:%{p|pg:-Y P,%R/usr/ccs/lib/libp:%R/usr/lib/libp:%R/usr/ccs/lib:%R/usr/lib} \
-             %{!p:%{!pg:-Y P,%R/usr/ccs/lib:%R/usr/lib}}}}"
+     %{!YP,*:%{p|pg:-Y P,/usr/ccs/lib/libp:/usr/lib/libp:/usr/ccs/lib:/lib:/usr/lib:/usr/gnu/lib:/usr/gcc/4.5/lib -R /usr/ccs/lib/libp:/usr/lib/libp:/usr/ccs/lib:/lib:/usr/lib:/usr/gnu/lib:/usr/gcc/4.5/lib} \
+             %{!p:%{!pg:-Y P,/usr/ccs/lib:/lib:/usr/lib:/usr/gnu/lib:/usr/gcc/4.5/lib -R /usr/ccs/lib:/lib:/usr/lib:/usr/gnu/lib:/usr/gcc/4.5/lib}}}}"
 
 #undef LINK_ARCH32_SPEC
 #define LINK_ARCH32_SPEC LINK_ARCH32_SPEC_BASE
