Index: exrenvmap/main.cpp
===================================================================
--- exrenvmap/main.cpp
+++ exrenvmap/main.cpp
@@ -44,6 +44,7 @@
 #include <ImfEnvmap.h>
 #include <iostream>
 #include <exception>
+#include <string.h>
 #include <stdlib.h>
 
 using namespace Imf;
Index: exrmaketiled/main.cpp
===================================================================
--- exrmaketiled/main.cpp
+++ exrmaketiled/main.cpp
@@ -44,7 +44,7 @@
 
 #include <iostream>
 #include <exception>
-#include <string>
+#include <string.h>
 #include <stdlib.h>
 
 using namespace Imf;
