Index: binutils-2.19.1/etc/Makefile.in
===================================================================
--- binutils-2.19.1.orig/etc/Makefile.in	2011-04-27 20:36:29.489244006 +0400
+++ binutils-2.19.1/etc/Makefile.in	2011-04-27 20:37:11.734345628 +0400
@@ -81,12 +81,6 @@
 
 install-info: info
 	$(SHELL) $(srcdir)/../mkinstalldirs $(DESTDIR)$(infodir)
-	if test ! -f standards.info; then cd $(srcdir); fi; \
-	if test -f standards.info; then \
-	  for i in standards.info*; do \
-	    $(INSTALL_DATA) $$i $(DESTDIR)$(infodir)/$$i; \
-	  done; \
-	fi
 	if test ! -f configure.info; then cd $(srcdir); fi; \
 	if test -f configure.info; then \
 	  for i in configure.info*; do \
